src/components/DataGridWithInfiniteScroll/DataGridWithInfiniteScroll.tsx

```diff
@@ -0,0 +1,862 @@
+ * @param {ASC} to sort data in ascending order
+ * @param {NONE} no sorting
+ */
+export enum SortOrder {
+    DESC = "descending",
+    ASC = "ascending ",
+    NONE = "none",
+}
+
+/**
+ * type for DataGridSort :
+ * @param {defaultSortOrder} if data in column by default sorted
+ * @param {sortFunction} function to perform sorting
+ * @param {isSorted} checks if column is currently sorted or not
+ * @param {hideSort} if true hides sort
+ */
+export type DataGridSort = {
+    defaultSortOrder: SortOrder;
+    sortFunction: (
+        rows: DataGridRow[],
+        order: SortOrder,
+        columnName: string
+    ) => Promise<DataGridRow[]>;
+    isSorted?: boolean;
+    hideSort?: boolean;
+};
+
+/**
+ * type for datagrid row detail pane data :
+ * @param {title} title for detail pane
+ * @param {columnNames} ColumnNames is for detail panel column display
+ * @param {detailPaneContentJSON} detailPaneContentJSON  to fetch row detail pane contents in JSON
+ */
+export type DetailPaneData = {
+/**
+ * TODO:
+ * Type 'Any' will be changed to specific types while writing wrappers for props.
+ */
+    title?: any;
+    columnNames?: any;
+    detailPaneContentJSON: any;
+};
+
+/**
+ * @param {className} CSS class names
+ * @param {style} CSS styles
+ * @param {id} unique ID for datagrid
+ * @param {rows} rows data
+ * @param {columns} column details
+ * @param {selectionType} row selection type that is multi or single
+ * @param {rowType} Expandable or compact row type
+ * @param {isSorting} if true, sorting on datagrid will appear else not
+ * @param {isFilter} if true, Filter panel will appear else not
+ * @param {showColumnSelect} showColumnSelect is boolean value if true,column can be show or hide
+ * @param {detailDataProps} detailDataProps is to get the data for detail panel in key value format
+ * @param {detailPaneContent} detailPaneContent is to display funtion in detail panel instean of key value
+ * @param {showDetailPanel} if true, detail panel on datagrid will appear else not
+ * @param {defaultColumnHeader} defaultColumnHeader is to set one default column on Column Show Hide functionality to show that column on DataGrid
+ * @param {filterData} filterData is set of html components which can be shown in Filter Panel.
+ * @param {getInfiniteScrollData} getInfiniteScrollData to get the data from function calling the infinite data
+ * @param {fetchMoreData} fetchMoreData will fetch next set of record in Infinite Scroll once user is end of the page
+ * @param {datagridBodyHeight}  is fixed height for the body of the datagrid
+ * @param {isLoading} if true then show loading icon for expandable row
+ * @param {callbackForColumnName} callback function to send columnName which user had clicked for sorting
+ * @param {loadingMessage} message to be passed while the infinite scroll loads nect data
+
+ */
+
+/**
+ * TODO:
+ * Type 'Any' will be changed to specific types while writing wrappers for props.
```

Comment:
are u going to replace style? : any with specific type ? No right ?
So please add inline comment at proper places